### PR TITLE
docs: Correct `Cypress.minimatch()` examples.

### LINKED ISSUE #4286: `Cypress.minimatch` fails to match as the docs suggest
### Current behavior

Cypress.minimatch's behavior is either not correctly documented, or it is broken. I noticed this when trying to match routes—as suggested in the docs—with just the path.

Here is a test whose body is taken directly from the docs (see below):

```js
it("matches things...", () => {
  console.log( Cypress.minimatch( "http://localhost/users?_limit=3", "/users?_limit=+(3|5)" ) )
  console.log( Cypress.minimatch( "http://localhost/users?_limit=5", "/users?_limit=+(3|5)" ) )
  console.log( Cypress.minimatch( "http://localhost/users?_limit=7", "/users?_limit=+(3|5)" ) )
})
```

This produces the following in the browser console:
```
false
false
false
```

### Desired behavior

The result of the code above should be
```
true
true
false
```

It seems to me that the docs are not describing actual behavior for url matching.

From https://docs.cypress.io/api/commands/intercept#Glob-Pattern-Matching-URLs

> Glob pattern matching provides the necessary flexibility:
> ```js
> cy.intercept('/users')
> // matches all of these:
> //   https://prod.cypress.io/users
> //   https://staging.cypress.io/users
> //   http://localhost/users
> 
> cy.intercept('/users?_limit=+(3|5)')
> // matches all of these:
> //   https://prod.cypress.io/users?_limit=3
> //   http://localhost/users?_limit=5
> ```

And the examples in the minimatch section just below show how you can test/debug matches:

> Under the hood, Cypress uses the minimatch library for glob matching and provides access to it via the Cypress global. This enables you to test your pattern in the Test Runner browser console.
> 
> You can invoke the Cypress.minimatch with just two arguments - the URL (string) and the pattern (string), respectively - and if it yields true, then you have a match!
>
> ```js
> // executed in the Test Runner browser console:
> Cypress.minimatch('http://localhost/users?_limit=3', '/users?_limit=+(3|5)')
> // true
> Cypress.minimatch('http://localhost/users?_limit=5', '/users?_limit=+(3|5)')
> // true
> Cypress.minimatch('http://localhost/users?_limit=7', '/users?_limit=+(3|5)')
> // false
> ```

That last bit is where I got the code that I put into my test.

### Test code to reproduce

(Given above)

### Cypress Version

Tested in 9.1.0 and 9.1.1

### Other

I referenced this problem in cypress-io/cypress#17729 , though the OP in that issue was describing something different. One of the commenters seemed to think that the `url` match is strictly string-based by default. It certainly appears that Cypress behaves that way, but that seems to be at odds with the documentation.

In any case, the docs clearly assert falsely that the test I wrote (copy/pasted) should produce different results.

In addition, I actually had to add `console.log` to each of the `Cypress.minimatch` calls to get the results to show up in the console. The docs seem also to suggest that you don't need to actually invoke `console.log`... which is wrong.

### PR DESCRIPTION
**Pull request by emilyrohrbough**
Closes #4286

### CODE REVIEW

File: content/api/commands/intercept.md [hunk 1]
```diff
@@ -1451,21 +1452,26 @@ cy.intercept('/users?_limit=+(3|5)')
 ### Cypress.minimatch
 
 Under the hood, Cypress uses the [minimatch](/api/utilities/minimatch) library
-for glob matching and provides access to it via the `Cypress` global. This
-enables you to test your pattern in the Test Runner browser console.
+with the `{ matchBase: true}` option applied for glob matching and provides
```
Comment (tbiethman) [inline content/api/commands/intercept.md:1455]:
```suggestion
with the `{ matchBase: true }` option applied for glob matching and provides
```

File: content/api/commands/intercept.md [hunk 2]
```diff
@@ -1451,21 +1452,26 @@ cy.intercept('/users?_limit=+(3|5)')
...
+access to it via the `Cypress` global. This enables you to test your pattern in
+your spec or in Test Runner browser console.
```
Comment (tbiethman) [inline content/api/commands/intercept.md:1457]:
```suggestion
your spec or in the Test Runner browser console.
```

**Review by tbiethman — APPROVED** [review]

File: content/api/commands/intercept.md [hunk 3]
```diff
@@ -1451,21 +1452,26 @@ cy.intercept('/users?_limit=+(3|5)')
 ### Cypress.minimatch
 
 Under the hood, Cypress uses the [minimatch](/api/utilities/minimatch) library
```
Comment (AtofStryker) [inline content/api/commands/intercept.md:1454]:
```suggestion
Under the hood, `cy.intercept` uses the [minimatch](/api/utilities/minimatch) library
```

When I read this, I think every invocation of `Cypress.minimatch` adds `{ matchBase: true }` as an option and not just to `cy.intercept`, even with the above context. Is this clear for others or is it just me?

Comment (emilyrohrbough) [reply content/api/commands/intercept.md:1454]: Good point!

**Review by AtofStryker — APPROVED** [review]

File: content/api/commands/intercept.md [hunk 4]
```diff
@@ -1450,22 +1451,27 @@ cy.intercept('/users?_limit=+(3|5)')
...
-Cypress.minimatch('http://localhost/users?_limit=7', '/users?_limit=+(3|5)')
-// false
+expect(
+  Cypress.minimatch('http://localhost/users?_limit=3', '**/users?_limit=+(3|5)')
+).to.be.true
+expect(
+  Cypress.minimatch('http://localhost/users?_limit=5', '/users?_limit=+(3|5)', {
+    matchBase: true,
+  })
+).to.be.true
```
Comment (happycollision) [inline content/api/commands/intercept.md:1471]: Very interesting!

**Pull request merged**